### PR TITLE
Issue Overlay

### PR DESCRIPTION
**Pull request by basil-ahmed**
Issue Overlay

### CODE REVIEW

**Comment by swostikpati** [comment]
Reviewed PR

- Hardcoded dummy data is used. The student dashboard screen already made API requests to the backend to fetch the data (implemented by @hasiburratul). All that was required was to pass this data as props and then render it as issue details. But instead hardcoded data was used.
- Any updates/comments cannot be submitted
- Almost none of the buttons are implemented (atleast in a mock way making a request to a particular API or maybe changing the state variable).
- The page lacks almost any kind of styling - especially based on NYU styling guide (the color palette of which is avaiable as constants globally). 
image

File: front-end/src/components/student/IssueDetails.js [hunk 1]
```diff
@@ -1,44 +1,90 @@
...
+import { useParams, useNavigate } from 'react-router-dom';
+import './IssueDetails.css';
 
-// TODO: Issue details display
+const IssueDetails = () => {
+    const { index } = useParams();
+    const [issue, setIssue] = useState(null);
+    const navigate = useNavigate();
 
-// Dummy issue details
```
Comment (swostikpati) [inline front-end/src/components/student/IssueDetails.js:6]: hardcoded data

File: front-end/src/components/student/IssueDetails.js [hunk 2]
```diff
@@ -1,44 +1,90 @@
...
+              <div className="attachments">
+                  <h3>Attachments</h3>
+                  <ul>
+                      {attachments.map((attach, index) => <li key={index}>{attach}</li>)}
+                  </ul>
+              </div>
+          </aside>
+
+          <div className="footer-buttons">
+              <button>Reopen Issue</button>
```
Comment (swostikpati) [inline front-end/src/components/student/IssueDetails.js:83]: buttons have no onClick attribute attached to them

File: front-end/src/components/student/IssueDetails.css [hunk 3]
```diff
@@ -0,0 +1,65 @@
+.issue-details-container {
+    display: grid;
+    grid-template-rows: auto 1fr auto;
+    grid-template-columns: 2fr 1fr;
+    gap: 20px;
+    padding: 20px;
+    position: relative;
+}
+
+header {
```
Comment (swostikpati) [inline front-end/src/components/student/IssueDetails.css:10]: styling directly on global element

**Comment by hasiburratul** [comment]
No additional review. @swostikpati covered everything.

**Pull request closed without merging**